c7n/resolver.py

```diff
@@ -151,6 +163,7 @@ def get_contents(self):
             uri=self.data.get('url'),
             headers=self.data.get('headers', {})
         )
+        self._add_api_key(params['headers'])
```

Comment:
What do you think about moving `if` conditions out of the function? For example, call `_add_api_key` function only if `api_key_secret` is set?